src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -739,4 +756,74 @@ private RepairRun buildRepairRunFromRow(Row repairRunResult, long id){
         .build(id);
   }
 
+  @Override
+  public boolean takeLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(getLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.reaperInstanceId, ReaperApplication.getInstanceAddress())).one();
+    if (lwtResult.getBool("[applied]")) {
+      LOG.debug("Took lead on segment {}", segmentId);
+      return true;
+    }
+    
+    // Another instance took the lead on the segment
+    LOG.debug("Could not take lead on segment {}", segmentId);
+    return false;
+  }
+
+  @Override
+  public boolean renewLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(renewLeadOnSegmentPrepStmt.bind(ReaperApplication.reaperInstanceId, ReaperApplication.getInstanceAddress(), segmentId, ReaperApplication.reaperInstanceId)).one();
+    if (lwtResult.getBool("[applied]")) {
+      LOG.debug("Renewed lead on segment {}", segmentId);
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public void releaseLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(releaseLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.reaperInstanceId)).one();
+    LOG.debug("Released lead on segment {}", segmentId);
+  }
+
+  @Override
+  public void storeHostMetrics(HostMetrics hostMetrics) {    
+    session.execute(storeHostMetricsPrepStmt.bind(hostMetrics.getHostAddress(), hostMetrics.getPendingCompactions(), hostMetrics.hasRepairRunning(), hostMetrics.getActiveAnticompactions()));
+  }
+
+  @Override
+  public Optional<HostMetrics> getHostMetrics(String hostName) {
+    ResultSet result = session.execute(getHostMetricsPrepStmt.bind(hostName));
+    for(Row metrics:result) {
+      return Optional.of(HostMetrics.builder().withHostAddress(hostName)
+                                  .withPendingCompactions(metrics.getInt("pending_compactions"))
+                                  .withHasRepairRunning(metrics.getBool("has_repair_running"))
+                                  .withActiveAnticompactions(metrics.getInt("active_anticompactions"))
+                                  .build());
+     }
+    
+    return Optional.absent();
+  }
+
+  @Override
+  public StorageType getStorageType() {
+    return StorageType.CASSANDRA;
+  }
+
+  @Override
+  public int countRunningReapers() {
+    ResultSet result = session.execute(getRunningReapersCountPrepStmt.bind());
+    int runningReapers = (int) result.one().getLong("nb_reapers");
+    LOG.debug("Running reapers = {}", runningReapers);
+    return runningReapers>0?runningReapers:1;
+  }
+
+  @Override
+  public void saveHeartbeat() {
+    DateTime now = DateTime.now();
+    // Send heartbeats every minute
+    if(now.minusSeconds(60).getMillis() >= lastHeartBeat.getMillis()) {
+      session.executeAsync(saveHeartbeatPrepStmt.bind(ReaperApplication.reaperInstanceId, ReaperApplication.getInstanceAddress()));
+      lastHeartBeat = now;
+    }
+  }
```

Comment:
with the schema using a TTL of 3 minutes this code's correct becomes dependent on the heartbeat.

the heartbeat could be strengthen, as it can breaks easily, see `ReaperApplication.scheduler`, by making the reaper instance crash if the scheduler calls fail (for any reason what-so-ever).